Framework/Core/Windows/Devices/I2c/I2cDevice.cs

```diff
@@ -0,0 +1,206 @@
+            }
+
+            if (deviceId != "I2C0")
+            {
+                // TODO: Is this the right exception?
+                //throw new InvalidOperationException();
+            }
+
+            return new I2cDevice((ushort)settings.SlaveAddress, settings.BusSpeed);
+        }
+
+        public void Write(byte[] writeBuffer)
+        {
+            WritePartial(writeBuffer);
+        }
+
+        public I2cTransferResult WritePartial(byte[] buffer)
+        {
+            lock (m_syncLock)
+            {
+                if (m_disposed)
+                {
+                    throw new ObjectDisposedException();
+                }
+
+                var transactions = new I2CDevice.I2CTransaction[] { I2CDevice.CreateWriteTransaction(buffer) };
+                return ExecuteTransactions(transactions);
+            }
+        }
+
+        public void Read(byte[] readBuffer)
+        {
+            ReadPartial(readBuffer);
+        }
+
+        public I2cTransferResult ReadPartial(byte[] buffer)
+        {
+            lock (m_syncLock)
+            {
+                if (m_disposed)
+                {
+                    throw new ObjectDisposedException();
+                }
+
+                var transactions = new I2CDevice.I2CTransaction[] { I2CDevice.CreateReadTransaction(buffer) };
+                return ExecuteTransactions(transactions);
+            }
+        }
+
+        public void WriteRead(byte[] writeBuffer, byte[] readBuffer)
+        {
+            WriteReadPartial(writeBuffer, readBuffer);
+        }
+
+        public I2cTransferResult WriteReadPartial(byte[] writeBuffer, byte[] readBuffer)
+        {
+            lock (m_syncLock)
+            {
+                if (m_disposed)
+                {
+                    throw new ObjectDisposedException();
+                }
+
+                var transactions = new I2CDevice.I2CTransaction[] {
+                    I2CDevice.CreateWriteTransaction(writeBuffer),
+                    I2CDevice.CreateReadTransaction(readBuffer) };
+                return ExecuteTransactions(transactions);
+            }
+        }
+
+        public void Dispose()
+        {
+            lock (m_syncLock)
+            {
+                if (!m_disposed)
+                {
+                    Dispose(true);
+                    GC.SuppressFinalize(this);
+                    m_disposed = true;
+                }
+            }
+        }
+
+        // Private methods
+
+        private I2cTransferResult ExecuteTransactions(I2CDevice.I2CTransaction[] transactions)
+        {
+            uint bytesRequested = 0;
+            foreach (var transaction in transactions)
+            {
+                bytesRequested += (uint)transaction.Buffer.Length;
+            }
+
+            I2cTransferResult result;
+
+            lock (s_deviceLock)
+            {
+                // TODO: How long should our transaction timeout be? 15ms is the max in UWP, but too short for us.

```

Comment:
Why is it too short? Either way the hardcoded value should be converted to a named const (NOTE: the C# language and .NET runtime require that any reference to the const be replaced with the actual literal value inline in the IL code and therefore the const declaration has no runtime overhead compared to hard coding the value everywhere.)